superset/common/query_object.py

```diff
@@ -116,24 +117,24 @@ def __init__(
         order_desc: bool = True,
         extras: Optional[Dict[str, Any]] = None,
         columns: Optional[List[str]] = None,
-        orderby: Optional[List[List[str]]] = None,
+        orderby: Optional[List[OrderBy]] = None,
         post_processing: Optional[List[Optional[Dict[str, Any]]]] = None,
         is_rowcount: bool = False,
         **kwargs: Any,
     ):
+        columns = columns or []
+        groupby = groupby or []
+        extras = extras or {}
+        annotation_layers = annotation_layers or []
+
         self.is_rowcount = is_rowcount
         self.datasource = None
         if datasource:
             self.datasource = ConnectorRegistry.get_datasource(
                 str(datasource["type"]), int(datasource["id"]), db.session
             )
         self.result_type = result_type
-        annotation_layers = annotation_layers or []
         self.apply_fetch_values_predicate = apply_fetch_values_predicate or False
-        metrics = metrics or []
```

Comment:
`metrics` is allowed to be `None` so we don't need a fallback here. `metrics is None` indicates there's no need for aggregation.

Reply:
I think `[]` and `None` are used fairly interchangeably in the codebase, but I agree that setting `x = None` for cases where `x` is not relevant communicates intent better.